.github/workflows/publish.yml

```diff
@@ -1,29 +1,86 @@
-name: brew pr-pull
+name: Publish Bottles
 on:
-  pull_request_target:
-    types:
-      - labeled
+  pull_request:
+    paths:
+      - Formula/*.rb
+
 jobs:
-  pr-pull:
-    if: contains(github.event.pull_request.labels.*.name, 'pr-pull')
-    runs-on: ubuntu-22.04
+  build-bottles:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Set up Homebrew
         uses: Homebrew/actions/setup-homebrew@master
 
+      - run: brew test-bot --only-cleanup-before
+
+      - run: brew test-bot --only-setup
+
+      - run: brew test-bot --only-tap-syntax
+
+      - run: brew test-bot --only-formulae
+
+      - name: Upload bottles as artifact
+        uses: actions/upload-artifact@main
+        with:
+          name: bottles
+          path: '*.bottle.*'
+
+  brew-bottle:
+    runs-on: ubuntu-latest
+    needs: build-bottles
+    steps:
+      - name: Set up Homebrew
+        run: |
```

Comment:
Does the `Homebrew/actions/setup-homebrew@master` action not do everything expected to set up Homebrew here?

Reply:
That actions sets up the repository in a bit of a [weird way](https://github.com/Homebrew/actions/blob/0c611abec1fc679030907286a2c084cc92fb8f2e/setup-homebrew/main.sh#L149-L153). And it's not configurable at all. Since we want to push commits back to the PR, we need to be able to configure it.

But it might work if we run `actions/checkout` after that action... 🤔

Reply:
Testing this idea [here](https://github.com/phylum-dev/homebrew-cli/actions/runs/3432142113)

Reply:
It works. Code updated